Packs/Box/Integrations/BoxEventsCollector/BoxEventsCollector.py

```diff
@@ -0,0 +1,227 @@
+
+    def _create_authorization_body(self):
+        claims = Claims(
+            client_id=self.box_credentials.boxAppSettings.clientID,
+            id=self.box_credentials.enterpriseID,
+        )
+
+        decrypted_private_key = _decrypt_private_key(
+            self.box_credentials.boxAppSettings.appAuth
+        )
+        assertion = jwt.encode(
+            payload=claims.dict(),
+            key=decrypted_private_key,
+            algorithm='RS512',
+            headers={
+                'kid': self.box_credentials.boxAppSettings.appAuth.publicKeyID
+            },
+        )
+        body = {
+            'grant_type': 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+            'assertion': assertion,
+            'client_id': self.box_credentials.boxAppSettings.clientID,
+            'client_secret': self.box_credentials.boxAppSettings.clientSecret,
+        }
+        return body
+
+
+class BoxGetEvents(IntegrationGetEvents):
+    client: BoxEventsClient
+
+    def get_last_run(self: Any) -> dict:  # type: ignore
+        demisto.debug(f'setting {self.client.request.params.stream_position=}')
+        return {'stream_position': self.client.request.params.stream_position}
+
+    def _iter_events(self):
+        self.client.authenticate()
+        demisto.debug('authenticated successfully')
+        # region First Call
+        events = self.client.call(self.client.request).json()
+        # endregion
+        # region Yield Response
+        while True:  # Run as long there are logs
+            self.client.set_request_filter(events['next_stream_position'])
+            demisto.debug(
+                f'setting then next request filter {events["next_stream_position"]=}'
+            )
+            if not events['entries']:
+                break
+            yield events['entries']
+            # endregion
+            # region Do next call
+            events = self.client.call(self.client.request).json()
+            # endregion
+
+
+def _decrypt_private_key(app_auth: AppAuth):
+    """
+    Attempts to load the private key as given in the integration configuration.
+
+    :return: an initialized Private key object.
+    """
+    try:
+        key = load_pem_private_key(
+            data=app_auth.privateKey.encode('utf8'),
+            password=app_auth.passphrase.encode('utf8'),
+            backend=default_backend(),
+        )
+    except (
+        TypeError,
+        ValueError,
+        exceptions.UnsupportedAlgorithm,
+    ) as exception:
+        raise DemistoException(
+            'An error occurred while loading the private key.', exception
+        )
+    return key
+
+
+def main(command: str, demisto_params: dict):
+    box_credentials = BoxCredentials.parse_raw(
+        demisto_params['credentials_json']['password']
+    )
+    request = BoxEventsRequest(
+        params=BoxEventsParams.parse_obj(demisto_params),
+        **demisto_params,
```

Comment:
Why both?

Reply:
because I don't have params in params. I can add it to params but it seems less explisit.